assets/js/components/higherorder/withdata.js

```diff
@@ -219,34 +221,32 @@ const withData = (
 
 			// Resolve all selectedData.
 			each( selectData, ( data ) => {
+				const { type, identifier } = data || {};
 				// Handle single contexts, or arrays of contexts.
-				if ( Array.isArray( data.context ) ) {
-					each( data.context, ( acontext ) => {
-						/**
-						 * Request data for the context.
-						 */
-						addFilter( `googlesitekit.module${ acontext }DataRequest`,
-							`googlesitekit.data${ acontext }`, ( moduleData ) => {
-								data.callback = ( returnedData ) => {
-									handleReturnedData( returnedData, data );
-								};
-								moduleData.push( data );
-								return moduleData;
-							} );
```

Comment:
This was previously duplicated for both the array and non-array cases. It's been consolidated into a single block using `castArray`.